src/workerd/jsg/modules.c++

```diff
@@ -306,7 +306,10 @@ v8::Local<v8::Module> compileEsmModule(
     jsg::Lock& js,
     kj::StringPtr name,
     kj::ArrayPtr<const char> content,
-    ModuleInfoCompileOption option) {
+    ModuleInfoCompileOption option,
+    const ModuleRegistryObserver& observer) {
+  auto compilationObserver = observer.onEsmCompilationStart(js.v8Isolate, name, option);
```

Comment:
Adding a comment about how `compilationObserver` is expected to be "used" here would be helpful to future generations.